src/common/compress/lz4/compress.c

```diff
@@ -0,0 +1,286 @@
+/***********************************************************************************************************************************
+LZ4 Compress
+
+Developed using against version r131 using the documentation in https://github.com/lz4/lz4/blob/r131/lib/lz4frame.h.
+***********************************************************************************************************************************/
+#include "build.auto.h"
+
+#ifdef HAVE_LIBLZ4
+
+#include <stdio.h>
+#include <lz4frame.h>
+#include <string.h>
+
+#include "common/compress/lz4/common.h"
+#include "common/compress/lz4/compress.h"
+#include "common/debug.h"
+#include "common/io/filter/filter.intern.h"
+#include "common/log.h"
+#include "common/memContext.h"
+#include "common/object.h"
+
+/***********************************************************************************************************************************
+Older versions of lz4 do not define the max header size.  This seems to be the max for any version.
+***********************************************************************************************************************************/
+#ifndef LZ4F_HEADER_SIZE_MAX
+    #define LZ4F_HEADER_SIZE_MAX                                    19
+#endif
+
+/***********************************************************************************************************************************
+Filter type constant
+***********************************************************************************************************************************/
+STRING_EXTERN(LZ4_COMPRESS_FILTER_TYPE_STR,                         LZ4_COMPRESS_FILTER_TYPE);
+
+/***********************************************************************************************************************************
+Object type
+***********************************************************************************************************************************/
+#define LZ4_COMPRESS_TYPE                                           Lz4Compress
+#define LZ4_COMPRESS_PREFIX                                         lz4Compress
+
+typedef struct
+{
+    MemContext *memContext;                                         // Context to store data
+    LZ4F_compressionContext_t context;                              // LZ4 compression context
+    LZ4F_preferences_t prefs;                                       // Preferences -- just compress level set
+    IoFilter *filter;                                               // Filter interface
+    Buffer *buffer;                                                 // For when the output buffer can't accept all compressed data
+    bool first;                                                     // Is this the first call to process?
+
+    bool inputSame;                                                 // Is the same input required on the next process call?
+    bool flushing;                                                  // Is input complete and flushing in progress?
```

Comment:
No -- just written at different times.  We can make the same if it helps -- probably flushing is better.

Reply:
For anyone trying to write a new filter, it would be helpful to be consistent. I don't want to make more work, so if if it helps, to me when we use flushing it that means we are in the process of doing that action, and if we say flush, to me that means we expect that action to happen.